File: imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/ImageFileLoader.java
package com.esafirm.imagepicker.features.fileloader;

import android.content.Context;

import com.esafirm.imagepicker.features.common.ImageLoaderListener;

import java.io.File;
import java.util.ArrayList;

public interface ImageFileLoader {

    void loadDeviceImages(
            final boolean isFolderMode,
            final boolean onlyVideo,
            final boolean includeVideo,
            final boolean includeAnimation,
            final ArrayList<File> excludedImages,
            final ImageLoaderListener listener
    );

    void abortLoadImages();
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/ImageFileLoader.kt
package com.esafirm.imagepicker.features.fileloader

import com.esafirm.imagepicker.features.ImagePickerConfig
import com.esafirm.imagepicker.features.common.ImageLoaderListener

interface ImageFileLoader {
    fun loadDeviceImages(config: ImagePickerConfig, listener: ImageLoaderListener)
    fun abortLoadImages()
}